src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/ReplicationChecker.java

```diff
@@ -0,0 +1,105 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.services.operation;
+
+import lombok.extern.slf4j.Slf4j;
+import org.awaitility.core.ConditionTimeoutException;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.task.SimpleAsyncTaskExecutor;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.TaskOperationRequest;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskOperationType;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.response.TaskOperationResponse;
+import uk.gov.hmcts.reform.wataskmanagementapi.entity.TaskHistoryResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.entity.TaskResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.services.CFTTaskDatabaseService;
+import uk.gov.hmcts.reform.wataskmanagementapi.services.MIReportingService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+
+@Slf4j
+@Component
+@Profile("replica | preview")
+public class ReplicationChecker implements TaskOperationPerformService {
+    private static final String TASK_REPLICATION_ERROR = "TASK_REPLICATION_ERROR: Task replication not fount for [{}]";
+    private final CFTTaskDatabaseService cftTaskDatabaseService;
+    private final MIReportingService miReportingService;
+
+    public ReplicationChecker(CFTTaskDatabaseService cftTaskDatabaseService, MIReportingService miReportingService) {
+        this.cftTaskDatabaseService = cftTaskDatabaseService;
+        this.miReportingService = miReportingService;
+    }
+
+    @Override
+    public TaskOperationResponse performOperation(TaskOperationRequest taskOperationRequest) {
+        if (taskOperationRequest.getOperation().getType()
+            .equals(TaskOperationType.PERFORM_REPLICATION_CHECK)) {
+            return performReplicationCheck();
+        }
+        return new TaskOperationResponse();
+    }
+
+    public TaskOperationResponse performReplicationCheck() {
+        List<TaskResource> lastUpdatedTasks = cftTaskDatabaseService.findLastFiveUpdatedTasks();
+        SimpleAsyncTaskExecutor executor = new SimpleAsyncTaskExecutor();
+        List<TaskResource> notReplicated = new ArrayList<>();
+        List<Future<?>> checks = new ArrayList<>();
+
+        for (TaskResource task : lastUpdatedTasks) {
+            Future<?> future = executor.submit(() -> {
+                try {
+                    await()
+                        .pollInterval(1, SECONDS)
+                        .atMost(20, SECONDS)
+                        .until(checkTaskHistory(task));
+                } catch (ConditionTimeoutException ex) {
+                    replicationLog(task);
```

Comment:
Could log the task id not replicated here?

Reply:
Task id is loged in the same log string.
https://github.com/hmcts/wa-task-management-api/blob/ec056aadc929b845dd6da5d71621a0ce967f62b5/src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/ReplicationChecker.java#L91